src/ESPAsyncWebServer.h

```diff
@@ -627,8 +627,8 @@ typedef std::function<void(AsyncWebServerRequest* request, uint8_t* data, size_t
 class AsyncWebServer {
   protected:
     AsyncServer _server;
-    std::list<std::shared_ptr<AsyncWebRewrite>> _rewrites;
-    std::list<std::unique_ptr<AsyncWebHandler>> _handlers;
+    std::vector<std::shared_ptr<AsyncWebRewrite>> _rewrites;
+    std::vector<std::unique_ptr<AsyncWebHandler>> _handlers;
```

Comment:
Will decrease memory usage using array at the expense of slower insertion and removal which is acceptable for rewrites and handlers which are not often added or removed